stan/math/prim/err/check_not_nan.hpp

```diff
@@ -49,7 +20,8 @@ struct not_nan<T_y, true> {
 template <typename T_y>
 inline void check_not_nan(const char* function, const char* name,
                           const T_y& y) {
-  internal::not_nan<T_y, is_vector_like<T_y>::value>::check(function, name, y);
+  auto is_good = [](const auto& y) { return !std::isnan(y); };
```

Comment:
Eigen has a ```.hasNan()``` [member function](https://eigen.tuxfamily.org/dox/classEigen_1_1DenseBase.html#ab13d158c900560d3e1b25d85d2d33dd6) that you could use